internal/scheduler/database/db_pruner.go

```diff
@@ -0,0 +1,103 @@
+package database
+
+import (
+	ctx "context"
+	"time"
+
+	"github.com/jackc/pgx/v4"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/util/clock"
+)
+
+// PruneDb removes completed jobs (and related runs and errors) from the database if their `lastUpdateTime`
+// is more than `keepAfterCompletion` in the past.
+// Jobs are deleted in batches across transactions. This means that if this job fails midway through, it still
+// may have deleted som jobs.
+// The function will run until the supplied context is cancelled.
+func PruneDb(ctx ctx.Context, db *pgx.Conn, batchLimit int, keepAfterCompletion time.Duration, clock clock.Clock) error {
+	start := time.Now()
```

Comment:
Should this be `clock.Now`?

Reply:
So I don't know if there's a clear answer to this.  It doesn't *have* to be `clock.Now` because this variable is only used to log out how long the migration took to run and we don't need to manipulate that in unit tests.  Basically the tradeoff is:
- If we use `clock.Now` the benefit  is that it means that all times in the function are consistent and reduces the risk that someone will  start using e.g. `start` for something else in future and then get weird things happening in tests.
- The disadvantage is that the log lines saying how long each function took to run will be nonsense when we're running tests.

Reply:
I spoke to @JamesMurkin and we came to an executive decision that leaving it as is was best- please shout if you disagree!